### DISCUSSION TITLE
max vs poolSize

### DISCUSSION THREAD

**Discussion by mprasu23** [vitaly-t/pg-promise#860] [General]
I am not able to understand the exact difference between the **max** and **poolSize** configuration parameters. 

As per this [pg-promise-wiki](https://github.com/vitaly-t/pg-promise/wiki/Connection-Syntax) **max** is the maximum number of clients the pool should contain.But there is no clear description of the poolSize. 

```
    const config = {
      host: 'HOST,
      database: 'DATABASE_NAME',
      user: 'POSTGRES_USERNAME',
      password: 'POSTGRES_PASSWORD',
      port: 'POSTGRES_PORT',
      poolSize: 10,
      max: 200
    }

```

As per my understanding **poolSize** is the maximum number of pools and **max** is the number of connections/clients per pool ?

**Comment by vitaly-t** [comment]
`poolSize` is legacy, `max` is new. Both work the same, presently, but you should just use `max`.

Currently, it works like [this](https://github.com/brianc/node-postgres/blob/a4ef6ce38c1e04bad2215312b1c79e64654cc857/packages/pg-pool/index.js#L84):

```ts
this.options.max = this.options.max || this.options.poolSize || 10
```

**Reply by mprasu23** [reply]
Thanks @vitaly-t for the clarification.

**Reply by mprasu23** [reply]
@vitaly-t  is **max** supported from any specific version of pg-promise ?

**Reply by vitaly-t** [reply]
From a while back. Just follow the [Connection Syntax](https://github.com/vitaly-t/pg-promise/wiki/Connection-Syntax) :wink:

**Reply by vitaly-t** [reply]
Option `poolSize` no longer exists in pg-promise v11.

**Comment by vitaly-t** [comment]
@mprasu23 You haven't accepted the answer. Any reason?
